### PR TITLE
Use builder pattern for the configuration

### PR DESCRIPTION
**Pull request by irevoire**
# Pull Request

## Related issue
Fixes https://github.com/meilisearch/arroy/issues/91

## What does this PR do?
- Provide a builder for querying the reader and building the trees

### CODE REVIEW

File: src/writer.rs [hunk 1]
```diff
@@ -27,6 +27,81 @@ use crate::{
...
+    }
+
+    /// Generates a forest of `n_trees` trees.
+    ///
+    /// More trees give higher precision when querying at the cost of more disk usage.
+    /// After calling build, no more items can be added.
+    ///
+    /// This function is using rayon to spawn threads. It can be configured
+    /// by using the [`rayon::ThreadPoolBuilder`] and the
+    /// [`rayon::ThreadPool::install`] to use it.
```
Comment (Kerollmops) [inline src/writer.rs:88]:
```suggestion
    /// [`rayon::ThreadPool::install`].
```

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Could you also update the README to change
> with the interface of the [Annoy Python library](https://github.com/spotify/annoy/#full-python-api) to search for vectors in space that are close to a given query vector

By 

> with an interface close to the [Annoy Python library](https://github.com/spotify/annoy/#full-python-api) to search for vectors in space that are near a targeted vector

File: src/writer.rs [hunk 2]
```diff
@@ -27,6 +27,81 @@ use crate::{
...
+    /// ```
+    pub fn split_after(&mut self, split_after: usize) -> &mut Self {
+        self.inner.split_after = Some(split_after);
+        self
+    }
+
+    /// Generates a forest of `n_trees` trees.
+    ///
+    /// More trees give higher precision when querying at the cost of more disk usage.
+    /// After calling build, no more items can be added.
```
[the remark this answers is not in the document]
Comment (irevoire) [reply src/writer.rs:85]: No, I removed it

**Review by Kerollmops — APPROVED** [review]
Thank you very much ❤️

**Pull request merged**